shop4All-backend/src/main/java/com/example/shop4All_backend/services/UserService.java

```diff
@@ -76,36 +82,64 @@ public void initRolesandUser() {
         sellerUser.setUserPassword(getEncodedPassword("seller"));
         sellerUser.setRole(Role.SELLER);
         sellerUser.setUserIsValid(true);
+        sellerUser.setUserCompanyName("COMPANY2024");
         userRepo.save(sellerUser);
+
+        Category category = new Category();
+        category.setCategoryName("categoryName1");
+        category.setCategoryDescription("categoryDesc1");
+        categoryRepo.save(category);
     }
 
     public String getEncodedPassword(String password) {
         return passwordEncoder.encode(password);
     }
 
+    //pattern to validate the email format
     private void validateEmail(String email) {
         String emailRegex = "^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+$";
         Pattern pattern = Pattern.compile(emailRegex);
         if (!pattern.matcher(email).matches()) {
-            throw new RequestException("Invalid email format");
+            throw new RegisterException("Invalid email format");
         }
     }
 
+    //pattern to validate the password format
     private void validatePassword(String password) {
         if (password.length() < 8) {
-            throw new RequestException("Password must be at least 8 characters long");
+            throw new RegisterException("Password must be at least 8 characters long");
         }
         if (!password.matches(".*[A-Z].*")) {
-            throw new RequestException("Password must contain at least one uppercase letter");
+            throw new RegisterException("Password must contain at least one uppercase letter");
         }
         if (!password.matches(".*[a-z].*")) {
-            throw new RequestException("Password must contain at least one lowercase letter");
+            throw new RegisterException("Password must contain at least one lowercase letter");
         }
         if (!password.matches(".*\\d.*")) {
-            throw new RequestException("Password must contain at least one digit");
+            throw new RegisterException("Password must contain at least one digit");
         }
         if (!password.matches(".*[!@#$%^&*].*")) {
-            throw new RequestException("Password must contain at least one special character (!@#$%^&*)");
+            throw new RegisterException("Password must contain at least one special character (!@#$%^&*)");
+        }
+    }
```

Comment:
Instead of having 4 ifs with some parts of a pattern, can you create a constant variable with all those parts of a pattern in a single big pattern that will be served as a Regex pattern for your desired password format and then check if the password matches the pattern and handle the potential exceptions? You can do something similar to the logic from the validateEmail() method.